src/FlaUI.Core/AutomationElements/AutomationElementExtensions.cs

```diff
@@ -73,5 +74,32 @@ public static partial class AutomationElementExtensions
             }
             return self;
         }
+
+        /// <summary>
+        /// Clicks in chosen element, creates instance of target object and returns it.
+        /// </summary>
+        /// <typeparam name="T">Type of target object to create.</typeparam>
+        public static T Click<T>(this AutomationElement element, TimeSpan? timeout = null, TimeSpan? interval = null) where T : AutomationElement
+        {
+            Retry.WhileException(() =>
+                {
+                    element.Click();
+                    Wait.UntilInputIsProcessed();
```

Comment:
btw. Click is calling this method internally already so I guess it could be skipped